### ISSUE TITLE
Shorten each module for easier code navigation and search

### ISSUE THREAD

**Issue by curtisma** [niftylab/laygo2#8]
Currently laygo2/object/grid.py is 3421 lines long.  This makes it harder to navigate and search the code.

I suggest that each class in this module be split into its own submodule.  Hopefully, this will allow most of the modules to be <200 lines long. 

So, for example, the "CircularMapping" class would be moved 
from "laygo2/object/grid.py"
to "laygo2/object/grid/CircularMapping.py"

**Comment by jdhan** [comment]
Yeah, I think it is a good idea to split long code into several files. We haven't done that so far as we wanted to keep similar things in one file. Perhaps some of the large files (physical.py, template.py, and grid.py) are better to be separated.
